build-api/api.py

```diff
@@ -1,10 +1,80 @@
-from flask import Flask
+import os
+import shutil
+import subprocess
+
+from flask import Flask, jsonify, render_template, request
 
 app = Flask(__name__)
 
-@app.route("/build")
+@app.route('/build', methods=['POST'])
 def build():
-    return "Hello World!"
+    json = request.get_json()
+    # an error is returned at this point if the json is bad
+    repo_name = json.get('repo_name', None)
+    repo_owner = json.get('repo_owner', None)
+    git_hash = json.get('git_hash', None)
+
+    # TODO - rsync final build results to the 'path' location
+    path = json.get('path', None)
+
+    if repo_name and repo_owner and git_hash and path:
+        tmp_dir = 'tmp/'
```

Comment:
Lets go a head and make this an environment variable. It will most likely be `/var/www` permanently, but adding some flexibility is probably a good idea.